imap_processing/mag/l1a/mag_l1a_data.py

```diff
@@ -336,6 +336,10 @@ def to_signed16(n: int) -> int:
         primary_vectors = []
         secondary_vectors = []
 
+        # To avoid overflows, we need to cast the potentially 8 bit signed integers to
+        # int32 before the bitshifting operations below.
+        vector_data = vector_data.astype(np.int32)
```

Comment:
@maxinelasp, I'm not sure how you want to handle this, so let me know if you want something else. You are calling `astype(int32)` in the code calling section, but in the tests you pass in a `frombuffer()` call which is only int8. We could get rid of the type-cast in the calling routine and rely on this, or we could immediately cast the frombuffer in the test too if you'd prefer.

Reply:
If this change passes the tests then I'm ok with it. The bitshifting is unfortunately complicated with different types, but I am also looking to rework the types to be more consistent in my newest change, so that it works with the uncompressed and compressed algorithms.

Reply:
> Is the problem that the 8 bit ints are signed?

The problem is here:
https://github.com/IMAP-Science-Operations-Center/imap_processing/blob/4a2642b09956be882f278b38175660a4b2e670da/imap_processing/mag/l1a/mag_l1a_data.py#L349-L352

where you are doing math with `0xFFFF` which is out of range for those types, so numpy2+ is warning and saying we need to be explicit with what types we want to do the math on.

Within the code you are already casting here:
https://github.com/IMAP-Science-Operations-Center/imap_processing/blob/4a2642b09956be882f278b38175660a4b2e670da/imap_processing/mag/l1a/mag_l1a.py#L185

But in the tests, you aren't:
https://github.com/IMAP-Science-Operations-Center/imap_processing/blob/4a2642b09956be882f278b38175660a4b2e670da/imap_processing/tests/mag/test_mag_l1a.py#L63-L65

So I was saying that we could also update this in those two locations if you wanted as well.

Reply:
Yeah, ok, in that case either change is fine, and I'll go back once the compression algorithm is done and make sure it's consistent with the types I'm using there.